icu-date.lua

```diff
@@ -199,7 +200,15 @@ function _M:parse(format, text, options)
 
   local text_uchar = string_to_uchar(text)
   position_ptr[0] = 0
-  return call_fn_check_status("udat_parseCalendar", format, self.cal, text_uchar, -1, position_ptr, status_ptr)
+  local res, err = call_fn_check_status("udat_parseCalendar", format, self.cal, text_uchar, -1, position_ptr, status_ptr)
+  if err ~= nil then
+    return nil, err
+  end
+  -- Workaround for https://unicode-org.atlassian.net/browse/ICU-21802
+  if position_ptr[0] ~= utf8.len(text) then
+    return nil, "String doesn't match format"
```

Comment:
Wouldn't it be better to return the same `Invalid status: "U_PARSE_ERROR". Result: "nil"` for consistency? If I get it right, we should expect exactly this error after library fix.

Reply:
Probably it makes sense. I wanted to highlight that there is a difference between "internal" libicu error and our workaround. But ok, fixed as you wish.